src/lib.rs

```diff
@@ -189,6 +189,64 @@ impl<'d> From<&'d *mut ffi::ibv_device> for Device<'d> {
     }
 }
 
+/// A Global unique identifier for ibv.
+///
+/// This struct acts as a rust wrapper for GUID value represented as `__be64` in
+/// libibverbs. We introduce this struct, because u64 is stored in host
+/// endianness, whereas ibverbs stores GUID in network order (big endian).
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Default, Copy, Clone, Debug, Eq, PartialEq, Hash)]
+#[repr(transparent)]
+pub struct Guid {
+    raw: [u8; 8],
+}
+
+impl Guid {
+    /// Upper 24 bits of the GUID are OUI (Organizationally Unique Identifier,
+    /// http://standards-oui.ieee.org/oui/oui.txt). The function returns OUI as
+    /// a 24-bit number inside a u64 number.
+    #[allow(dead_code)]
+    fn oui(&self) -> u32 {
+        let padded = vec![
+            0,
+            self.raw[0],
+            self.raw[1],
+            self.raw[2],
+        ];
+        u32::from_be_bytes(padded.try_into().unwrap())
```

Comment:
This should not be a `vec!`, but instead just an array directly:
```suggestion
        let padded = [
            0,
            self.raw[0],
            self.raw[1],
            self.raw[2],
        ];
        u32::from_be_bytes(padded)
```